sdk/monitor/azure-monitor-opentelemetry-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/pipeline/TelemetryItemSerialization.java

```diff
@@ -58,53 +96,58 @@ private static byte[] decode(byte[] rawBytes) {
         }
     }
 
-    public static List<ByteBuffer> serialize(List<TelemetryItem> telemetryItems) {
-        try {
-            if (logger.canLogAtLevel(LogLevel.VERBOSE)) {
-                StringWriter debug = new StringWriter();
-                try (JsonGenerator jg = mapper.createGenerator(debug)) {
-                    writeTelemetryItems(jg, telemetryItems, mapper);
-                }
-                logger.verbose("sending telemetry to ingestion service:{}{}", System.lineSeparator(), debug);
+    // split the byte array by newline character
+    public static List<byte[]> splitBytesByNewline(byte[] inputBytes) {
+        List<byte[]> lines = new ArrayList<>();
+        int start = 0;
+        for (int i = 0; i < inputBytes.length; i++) {
+            if (inputBytes[i] == '\n') {
+                byte[] line = new byte[i - start];
+                System.arraycopy(inputBytes, start, line, 0, i - start);
+                lines.add(line);
+                start = i + 1;
             }
+        }
+        // Add the last line (if any)
+        if (start < inputBytes.length) {
+            byte[] lastLine = new byte[inputBytes.length - start];
+            System.arraycopy(inputBytes, start, lastLine, 0, inputBytes.length - start);
+            lines.add(lastLine);
+        }
+        return lines;
+    }
 
-            ByteBufferOutputStream out = new ByteBufferOutputStream(byteBufferPool);
-
-            try (JsonGenerator jg = mapper.createGenerator(new GZIPOutputStream(out))) {
-                writeTelemetryItems(jg, telemetryItems, mapper);
+    // convert list of byte buffers to byte array
+    public static byte[] convertByteBufferListToByteArray(List<ByteBuffer> byteBuffers) {
```

Comment:
same for this method?

Reply:
yes, this just convert list<ByteBuffer> to byte[] before passing it to deserialize method.